pkg/daemon/config.go

```diff
@@ -255,52 +315,71 @@ func (conf *ptp4lConf) extractSynceRelations() *synce.Relations {
 	return r
 }
 
-func (conf *ptp4lConf) renderSyncE4lConf(ptpSettings map[string]string) (configOut string, relations *synce.Relations) {
+// RenderSyncE4lConf outputs synce4l config as string
+func (conf *Ptp4lConf) RenderSyncE4lConf(ptpSettings map[string]string) (configOut string, relations *synce.Relations) {
 	configOut = fmt.Sprintf("#profile: %s\n", conf.profile_name)
 	relations = conf.extractSynceRelations()
 	relations.AddClockIds(ptpSettings)
 	deviceIdx := 0
-	for _, section := range conf.sections {
-		configOut = fmt.Sprintf("%s\n%s", configOut, section.sectionName)
-		if strings.HasPrefix(section.sectionName, "[<") {
-			if _, found := section.options["clock_id"]; !found {
-				section.options["clock_id"] = relations.Devices[deviceIdx].ClockId
+
+	// Maintain consistent order of sections
+	sectionNames := make([]string, 0)
+	for sectionName := range conf.sectionOptions {
+		sectionNames = append(sectionNames, sectionName)
+	}
+	sort.Strings(sectionNames)
+
+	for _, sectionName := range sectionNames {
+		sectionOptions := conf.sectionOptions[sectionName]
+		configOut = fmt.Sprintf("%s\n%s", configOut, sectionName)
+		if strings.HasPrefix(sectionName, "[<") {
+			if _, found := conf.getPtp4lConfOptionOrEmptyString(sectionName, "clock_id"); !found {
+				conf.setPtp4lConfOption(sectionName, "clock_id", relations.Devices[deviceIdx].ClockId, true)
 				deviceIdx++
 			}
 		}
-		for k, v := range section.options {
+		for _, option := range sectionOptions {
+			k := option.key
+			v := option.value
 			configOut = fmt.Sprintf("%s\n%s %s", configOut, k, v)
 		}
 	}
 	return
 }
 
-func (conf *ptp4lConf) renderPtp4lConf() (configOut string, ifaces config.IFaces) {
+// RenderPtp4lConf outputs ptp4l config as string
+func (conf *Ptp4lConf) RenderPtp4lConf() (configOut string, ifaces config.IFaces) {
 	configOut = fmt.Sprintf("#profile: %s\n", conf.profile_name)
-	conf.mapping = nil
 	var nmea_source event.EventSource
 
-	for _, section := range conf.sections {
-		configOut = fmt.Sprintf("%s\n%s", configOut, section.sectionName)
+	// Maintain consistent order of sections
+	sectionNames := make([]string, 0)
+	for sectionName := range conf.sectionOptions {
+		sectionNames = append(sectionNames, sectionName)
+	}
+	sort.Strings(sectionNames)
```

Comment:
why this sorting? will this sorting change  order of interface provided ?

Reply:
updated